aws/resource_aws_eip.go

```diff
@@ -181,16 +181,20 @@ func resourceAwsEipRead(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
-	// Verify AWS returned our EIP
-	if len(describeAddresses.Addresses) != 1 ||
-		domain == "vpc" && *describeAddresses.Addresses[0].AllocationId != id ||
-		*describeAddresses.Addresses[0].PublicIp != id {
-		if err != nil {
-			return fmt.Errorf("Unable to find EIP: %#v", describeAddresses.Addresses)
+	var address *ec2.Address
+
+	// Sometimes AWS returns EIPs that *aren't* the EIP we're looking for
+	// so we loop over the returned addresses to see if it's in the list of results
+	for _, addr := range describeAddresses.Addresses {
+		if (domain == "vpc" && *addr.AllocationId == id) || *address.PublicIp == id {
```

Comment:
This new logic can cause Terraform to crash due to a typo:

```
=== RUN   TestAccAWSEIP_importEc2Classic

------- Stderr: -------
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x1ec009b]

goroutine 215 [running]:
github.com/terraform-providers/terraform-provider-aws/aws.resourceAwsEipRead(0xc420a6f8f0, 0x307b000, 0xc420592c00, 0x2f3d4e0, 0xc420952600)
	/opt/teamcity-agent/work/7be301907ad6c6ab/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_eip.go:189 +0x3bb
```

While we're fixing this, we should also switch the `*` dereferencing to the SDK provided `aws.StringValue()` function to further prevent panics:

```go
if (domain == "vpc" && aws.StringValue(addr.AllocationId) == id) || aws.StringValue(addr.PublicIp) == id {
```